app/src/main/java/com/stevesoltys/seedvault/restore/RestoreViewModel.kt

```diff
@@ -307,7 +307,22 @@ internal class RestoreViewModel(
          *   the current device. If no applicable datasets exist, restoreSets will be null.
          */
         override fun restoreSetsAvailable(restoreSets: Array<out RestoreSet>?) {
-            startRestore(this)
+            // this gets executed after we got the restore sets
+            // now we can start the restore of all available packages
+            restoreNextPackages()
+        }
+
+        private fun restoreNextPackages() {
+            // Packages are restored in chunks, or else transport.startRestore() in the framework's
+            // PerformUnifiedRestoreTask.java may fail due to an oversized Binder transaction,
+            // causing the entire restoration to fail.
+            val packagesChunk = packages.subList(0, min(packages.size, PACKAGES_PER_CHUNK))
```

Comment:
Isn't this always returning the same number of packages running into an infinite loop or what am I missin ghere?